test/unit/ClusterServiceTest.js

```diff
@@ -0,0 +1,404 @@
+
+        const addedMemberInfo = createMember({host:'127.0.0.2', port:5701, uuid:null, liteMember:false});
+        clusterService.handleMembersViewEvent(connectionRegistryStub, clusterUUID, 2, [member1, addedMemberInfo]);
+        expect(addedMembers).to.eql([addedMemberInfo]);
+    });
+
+    it('testRemoveListener', async function () {
+        let addedCount = 0;
+        const membershipListener = {
+            memberAdded: () => {
+                addedCount++;
+            },
+            memberRemoved: () => {}
+        };
+        const listenerUuid = clusterService.addMembershipListener(membershipListener);
+        expect(clusterService.removeMembershipListener(listenerUuid)).to.be.true;
+        const member1 = createMember({host:'127.0.0.1', port:5701, uuid:null, liteMember:false});
+        const clusterUUID = UuidUtil.generate();
+        // triggers initial event
+        clusterService.handleMembersViewEvent(connectionRegistryStub, clusterUUID, 1, [member1]);
+        const member2 = createMember({host:'127.0.0.2', port:5701, uuid:null, liteMember:false});
+        // triggers member added
+        clusterService.handleMembersViewEvent(connectionRegistryStub, clusterUUID, 2, [member2]);
+        // we have removed the listener. No event should be fired to our listener
+        expect(addedCount).to.be.equal(0);
+    });
+
+    it('testRemoveNonExistingListener', async function () {
+        expect(clusterService.removeMembershipListener(UuidUtil.generate())).to.be.false;
+    });
+
+    it('testGetMember', async function () {
+        const masterMember = createMember({host:'127.0.0.1', port:5701, uuid:null, liteMember:false});
+        const member2Uuid = UuidUtil.generate();
+        const member2 = createMember({host:'127.0.0.2', port:5701, uuid:member2Uuid, liteMember:false});
+        clusterService.handleMembersViewEvent(connectionRegistryStub, UuidUtil.generate(), 1, [masterMember, member2]);
+        const receivedMember = clusterService.getMember(member2Uuid.toString());
+        expect(member2).to.eql(receivedMember);
+    });
+
+    it('testGetMembers', async function () {
+        const masterMember = createMember({host:'127.0.0.1', port:5701, uuid:null, liteMember:false});
+        const liteMember = createMember({host:'127.0.0.2', port:5701, uuid:null, liteMember:true});
+        const dataMember = createMember({host:'127.0.0.3', port:5701, uuid:null, liteMember:false});
+        clusterService.handleMembersViewEvent(connectionRegistryStub, UuidUtil.generate(), 1,
+            [masterMember, liteMember, dataMember]);
+        expect([masterMember, liteMember, dataMember]).to.eql(clusterService.getMemberList());
+        expect([liteMember]).to.eql(clusterService.getMembers((selector) => {
+            return selector.liteMember;
+        }));
+        expect([masterMember, dataMember]).to.eql(clusterService.getMembers((selector) => {
+            return !selector.liteMember;
+        }));
+    });
```

Comment:
I think we can port testWaitInitialMembership as well. The corresponding method is waitForInitialMemberList

Reply:
Fixed